arangod/Aql/AttributeNamePath.cpp

```diff
@@ -32,29 +33,41 @@
 namespace arangodb {
 namespace aql {
 
-AttributeNamePath::AttributeNamePath(std::string attribute) {
-  path.emplace_back(std::move(attribute));
+AttributeNamePath::AttributeNamePath(
+    arangodb::ResourceMonitor& resourceMonitor) noexcept
+    : _path(ResourceUsageAllocator<MonitoredStringVector, ResourceMonitor>{
+          resourceMonitor}) {}
+
+AttributeNamePath::AttributeNamePath(std::string attribute,
+                                     arangodb::ResourceMonitor& resourceMonitor)
+    : _path(ResourceUsageAllocator<MonitoredStringVector, ResourceMonitor>{
+          resourceMonitor}) {
+  _path.emplace_back(std::move(attribute));
 }
 
-AttributeNamePath::AttributeNamePath(std::vector<std::string> p)
-    : path(std::move(p)) {}
+AttributeNamePath::AttributeNamePath(MonitoredStringVector p,
+                                     arangodb::ResourceMonitor& resourceMonitor)
+    : _path(ResourceUsageAllocator<MonitoredStringVector, ResourceMonitor>{
+          resourceMonitor}) {
+  _path = std::move(p);
+}
```

Comment:
Does the following simpler variant of this also do the trick?
```suggestion
AttributeNamePath::AttributeNamePath(MonitoredStringVector p)
    : _path(std::move(p)) {}
```

Reply:
let me check

Reply:
added: f150fbdf994be6154609841c78cca7b014a4e5cc